src/callbacks_step/save_restart_dg.jl

```diff
@@ -221,14 +270,6 @@ end
 function load_restart_file_parallel(mesh::Union{ParallelTreeMesh, ParallelP4estMesh,
```

Comment:
I did not change the `load_restart_file_on_root` since I could not test this on all of my system configurations, i.e., always `load_restart_file_parallel` would be called. When manually enforced, this part of the code errored:

```julia
        if read(attributes(file)["n_elements"]) != nelements(dg, cache)
            error("restart mismatch: number of elements in solver differs from value in restart file")
        end
```

Reply:
Did you try it with an environment without a LocalPreferenecs.toml file. `load_restart_file` will automatically call `load_restart_file_parallel` if it recognizes an MPI enabled HDF5 (which is configured by the LocalPreferences.toml file), i.e. it also calls the parallel function if you are not running the script with MPI.

Reply:
Yes, on my local machine I do not have such a file, and I tried this with the Trixi-only parallel 2D tree mesh.  I was also kinda confused.

Reply:
So `HDF5.has_parallel()` returns false?

Reply:
Ah, nowadays HDF5.jl is MPI-enabled by default also without custom MPI installation and configuration.

Reply:
I'm not quite sure anymore how. I think, when I implemented the parallel HDF5 support in Trixi.jl, HDF5.jl didn't come with parallel support out of the box. So it was easy to test. Now we need to tell HDF5.jl somehow *not* to use MPI.

Reply:
I managed to implement this now as well.

Reply:
Nice! Have you been able to test the function locally? If yes, what did you do to get `HDF5.has_parallel() == false`?

Reply:
Yes, I tested this on my machine. I just did the quick & dirty way of out-commenting the check for parallel HDF5 I/O :sweat_smile:

Reply:
Yeah, ok. That works, too :joy:
At least, it's tested in CI as reported by codecov. So this should be fine.

Reply:
Yeah the thing is I need to find the place in CI where this is green-lighted if I want to test this as well.